components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -424,6 +441,74 @@ public String exportApplication(String applicationId, Boolean exportSecrets) {
         }
     }
 
+    /**
+     * Export an application identified by the applicationId, as an XML string.
+     *
+     * @param applicationId ID of the application to be exported.
+     * @param exportSecrets If True, all hashed or encrypted secrets will also be exported.
+     * @return XML string of the application.
+     */
+    public TransferResource exportApplicationAsFile(String fileType, String applicationId, Boolean exportSecrets) {
+
+        if (StringUtils.isBlank(fileType)) {
+            throw new UnsupportedOperationException("No valid media type found");
+        }
+
+        ServiceProvider serviceProvider;
+        try {
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            serviceProvider = getApplicationManagementService().exportSPFromAppID(
+                    applicationId, exportSecrets, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            String msg = "Error exporting application with id: " + applicationId;
+            throw handleIdentityApplicationManagementException(e, msg);
+        }
+
+        if (serviceProvider == null) {
+            throw new UnsupportedOperationException("No valid service provider found");
```

Comment:
why are throwing 'UnsupportedOperationException' ?

Reply:
Yes it's redundant as any exception is caught in the try-catch block above. I have removed the if statement. 

It may have been added to handle a scenario in which the serviceProvider object could be null even if no exceptions were thrown.